Assets/Scripts/CameraManager.cs

```diff
@@ -153,19 +168,32 @@ public IEnumerator BlendToFarCameraForSeconds(float seconds)
         // now wait the given number of seconds
         yield return new WaitForSeconds(seconds);
 
+        // enable zooming from outside sources after we finishing zooming back in, 
+        //  awareness and such can affect the player on the close camera
+        OnBlendingComplete_Internal += ReenableZoomAfterBlend;
+
         // now blend back to the original camera
-        BlendTo(currentCamera, false);
+        BlendTo(currentCamera, OnBlendingStart_Internal, OnBlendingComplete_Internal);
     }
 
-    private void UpdateDefaultCameraDistance()
+    private void ReenableZoomAfterBlend(CinemachineVirtualCamera fromCamera, CinemachineVirtualCamera toCamera)
     {
-        defaultCameraDistanceInitialized = true;
+        OnBlendingComplete_Internal -= ReenableZoomAfterBlend;
+        canZoom = true;
+    }
 
-        // Update the our local activeCameraDefaultCameraDistance
-        CinemachineFramingTransposer blendToCameraFramingTransposer = GetActiveVirtualCamera().GetCinemachineComponent<CinemachineFramingTransposer>();
-        if (blendToCameraFramingTransposer)
+    private void UpdateDefaultCameraDistance(CinemachineVirtualCamera virtualCamera)
+    {
+        if (canZoom)
```

Comment:
Works fine for now & is clear to understand.

We can think about ways to improve this in the future; could a single variable (e.g. "playerLocked") be used to indicate that A) the player can't move, and B) nothing should be modifying the camera besides internal functions?  It could be stored in GameManager or PlayerManager, and we'll probably end up using it for other things as well.

Reply:
Ideally we can also move this "if (canZoom)" check further up (into "BlendTo"?) to prevent running through unnecessary lines of code.

Reply:
Thanks! 😄 

Well currently in `PlayerController` we have a public bool: `CanMove`, so I'm guessing that's the kind of thing you're referring to? Or are you be looking to add further functionality?

Also the "zoom" and the camera blending are separate things (as in, "zooming" only affects the current camera and how much farther/less it is zoomed in compared to it's original camera distance.
The blending is switching between two cameras, and the main issue was having the `AwarenessManager` constantly updating the "zoom" of the close camera, but then when blending to the far camera, we wanted no zoom to occur and just keep it's originally set camera distance.

So I'm not sure we can move the `if (canZoom)` check outside of this function? From what I can see.

Thanks for the quick review sir!